File: scraper/src/Helpers/Telegram/userUtils.py
from datetime import datetime
import os
from pymongo import MongoClient


def updateUserHistory(user_id, phone_number, result_id, platform):
    try:
        client = MongoClient(os.getenv("MONGO_URI"))
        db = client.get_database(os.getenv("MONGO_DB"))
        users_collection = db["users"]

        users_collection.update_one(
            {"_id": user_id},
            {
                "$addToSet": {
                    "searchHistory": {
                        "resultId": result_id,
                        "platform": platform,
                        "identifier": phone_number,
                        "timestamp": datetime.datetime.utcnow()
                    }
                }
            }
        )
        print(f"Search history updated for user {user_id}")
    except Exception as e:
        print(f"Error updating user search history: {e}")
    finally:
        client.close()

File: scraper/src/Helpers/userUtils.ts
import User from '../../../backend/models/userModel.js'

export const updateUserHistory = async (userId:any,startUrls:string[], resultId:any, platform:string) => {
    const user = await User.findById(userId);
    if (user) {
        user.searchHistory.push(resultId, startUrls, platform, startUrls);
        await user.save();
        console.log('Updated user search history successfully.');
    } else {
        console.error('User not found.');
        throw new Error('User not found');
    }
};
